streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -129,9 +129,12 @@ public void flush() {
 
     @Override
     public void close() {
-        flush();
-        underlying.close();
-        cache.close(cacheName);
+        try {
+            flush();
+        } finally {
+            underlying.close();
```

Comment:
Should we have a nested try-catch to make use `cache.close()` is called, too? Not sure how critical it is to ensure that the cache is closed?

Reply:
I had that, but I took it out for the same reason thinking it's not important to ensure the cache is closed.  But thinking about it some more, it doesn't hurt anything, so I'll put it back.